src/main/java/com/codahale/metrics/newrelic/NewRelicReporterFactory.java

```diff
@@ -0,0 +1,61 @@
+package com.codahale.metrics.newrelic;
+
+import com.codahale.metrics.MetricAttribute;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.SimpleMetricBatchSender;
+import com.newrelic.telemetry.metrics.MetricBatchSenderBuilder;
+import io.dropwizard.metrics.BaseReporterFactory;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.EnumSet;
+import java.util.Map;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@JsonTypeName("newrelic")
+public class NewRelicReporterFactory extends BaseReporterFactory {
+  @JsonProperty private String apiKey = null;
+
+  @JsonProperty private String overrideUri = null;
+
+  @JsonProperty private Map<String, Object> commonAttributes = null;
+
+  @Valid @NotNull @JsonProperty
+  private EnumSet<MetricAttribute> disabledMetricAttributes = EnumSet.noneOf(MetricAttribute.class);
+
+  @Override
+  @NotNull
+  public ScheduledReporter build(final MetricRegistry registry) {
+    final MetricBatchSenderBuilder metricBatchSender = SimpleMetricBatchSender.builder(apiKey);
```

Comment:
this variable should be called something like `metricBatchSenderBuilder`. Also, it doesn't need to be final.

Reply:
I mark all variable final unless they really do need to be modified. Additional safety and clarity.

Reply:
Understood that some people use this style. Let's keep with the style in line with the rest of the codebase, though, rather than change it just in one place. Thanks!